butterfree/extract/source.py

```diff
@@ -49,13 +49,22 @@ class Source(HookableComponent):
         temporary views regarding each reader and, after, will run the
         desired query and return a dataframe.
 
+        The `eager_evaluation` param forces Spark to apply the currently
+        mapped changes to the DataFrame. When this parameter is set to
+        False, Spark follows its standard behaviour of lazy evaluation.
+        Lazy evaluation can improve Spark's performance as it allows
+        Spark to build the best version of the execution plan.
+
     """
 
-    def __init__(self, readers: List[Reader], query: str) -> None:
+    def __init__(
+        self, readers: List[Reader], query: str = "", eager_evaluation: bool = True,
```

Comment:
By the docs, we have: "Lazy evaluation can improve Spark's performance as it allows Spark to build the best version of the execution plan.". Looks like that `False` as default is better, or am I wrong?

Reply:
Or is it `True` to maintain some compatibility?

Reply:
You're right, it's defaulted `True` so we can turn it off gradually in our pipelines.